File: php/foundation/FormParameters.php
<?php

class FormParameters {
	protected $parameterSet=null;
	protected $arraylength=0;
	
	public function __construct () {}
	
	public function put($key, $value) {
		if (is_null($this->parameterSet)) {
			$this->parameterSet=array();
			$this->arraylength=0;
		}
		$fp=new FormParameter($key, $value);
		$this->parameterSet[$this->arraylength]=$fp;
		++$this->arraylength;
	}
	
	public function getParameterSet() { return $this->parameterSet; }
	
	public function encodeParameters() {
		$data="";
		if ($this->parameterSet && count($this->parameterSet)>0) {
			for ($i=0; $i<count($this->parameterSet); $i++) {
				$param=$this->parameterSet[$i];
				$key=$param->getKey();
				$value=$param->getValue();
				if (isset($param) && isset($key) && isset($value)) { 
					if ($i>0) $data=$data."&\r\n";
					$data=$data.$key."=".urlencode($value);
				}
			}
		}
		return $data;
	}			
}

?>

File: Ruby/foundation/FormParameters.rb

require 'cgi'

require 'foundation/FormParameter'

class FormParameters 

	def initialize
		@parameterSet=Array.new()
	end
		
	def put(key, value)
		fp=FormParameter.new(key, value)
		@parameterSet[@parameterSet.length]=fp
	end
	
	def getParameterSet
		@parameterSet
	end
	
	def encodeParameters
	  rv=""
	  i=0
	  for fp in @parameterSet
	    if (fp.getKey!=nil) && (fp.getValue!=nil)
	      if i>0 
	        rv << "&"
	      end
	      rv << CGI::escape(fp.getKey.to_s)
	      rv << '='
	      rv << CGI::escape(fp.getValue.to_s)
	      i=i+1
	    end
	  end
	  return rv
	end
	
end
	
